### ISSUE TITLE
module.exports.setSettings for device not working anymore after upgrade to fw 0.10.0

### ISSUE THREAD

**Issue by gruijter** [athombv/homey#925]
**What is your Homey version (Settings → System)?**

0.10.0

**What did you try to achieve?**

*Please include detailed steps how we can reproduce the issue*
I use below code to programmatically update the settings from the magister app:

```
var settings = {
          period: result.period,
          type_group: result.type.description + " " + result.group.description
        };
        module.exports.setSettings( device_data, settings, function( err, settings ){
              // ... dunno what to do here, think nothing...
        });
```

**What was the result?**
When opening the devices settings screen the updated settings are incorrectly displayed (see screencapture). In Chrome console I get an error: ```TypeError: Cannot read property 'nl' of null```

![2016-09-29 1](https://cloud.githubusercontent.com/assets/19174120/18948882/2f9c970e-8639-11e6-9ca6-48025227e2e4.png)

...

**What did you expect as result?**
That the settings are updated correctly when using module.exports.setSettings
...

**Any other remarks**
This used to work correctly before fw 0.10.0
...

**Comment by WeeJeWel** [comment]
What are the exact values you're passing in `settings`? I cannot seem to reproduce your issue.

**Comment by gruijter** [comment]
@WeeJeWel  : the value I'm passing as 'settings' is this object:

{ period: '1617', type_group: '2 gymnasium 2ga' }

**Comment by gruijter** [comment]
Ok, I found the issue. Something must have changed in fw 0.10.x because this was not a problem before. But anyhow, I found out that an error was thrown on module.export.setSettings:
```{ [Error: invalid_settings_type] message: 'invalid_settings_type' }```

During pairing I gave the settings the following values:
{ period: null, type_group: null }

So when I tried to update the settings with a string, the problem is there. I have now changed the types to be a string already during pairing:
{ period: '', type_group: '' }

Problem solved....
